Services/Services_feeds/alternative_me_service_feed/alternative_me_feed.py

```diff
@@ -0,0 +1,128 @@
+
+    def _get_sleep_time_before_next_wakeup(self):
+        return commons_enums.TimeFramesMinutes[self.refresh_time_frame] * commons_constants.MINUTE_TO_SECONDS
+
+    async def _get_fear_and_greed_data(self, session: aiohttp.ClientSession, limit: typing.Optional[int] = 100) -> bool:
+        api_url = f"https://api.alternative.me/fng/?limit={limit}&format=json&date_format=us"
+        async with session.get(api_url) as response:
+            if response.status != 200:
+                self.logger.error(f"Alternative.me API request failed with status: {response.status}")
+                return False
+
+            fear_and_greed_data = await response.json()
+            data = fear_and_greed_data["data"]
+            self.data_cache[services_constants.ALTERNATIVE_ME_TOPIC_FEAR_AND_GREED] = [
+                AlternativeMeFearAndGreed(
+                    timestamp=datetime.datetime.strptime(entry["timestamp"], '%m-%d-%Y').timestamp(),
+                    value=float(entry["value"]),
+                    value_classification=entry["value_classification"]
+                ) for entry in data]
+            return True
+
+    def get_data_cache(self, current_time: float, key: typing.Optional[str] = None):
+        if self.data_cache is None:
+            return None
+
+        if key is None:
+            return self.data_cache
+
+        if key == services_constants.ALTERNATIVE_ME_TOPIC_FEAR_AND_GREED and self.data_cache.get(services_constants.ALTERNATIVE_ME_TOPIC_FEAR_AND_GREED) is not None:
+            return [item for item in self.data_cache.get(services_constants.ALTERNATIVE_ME_TOPIC_FEAR_AND_GREED) if item.timestamp <= current_time]
+        return None
+
+    async def _push_update_and_wait(self, session: aiohttp.ClientSession):
+        for topic in self.alternative_me_topics:
+            self.logger.debug(f"Fetching alternative.me {topic} topic data...")
+            result = False
+            if topic == services_constants.ALTERNATIVE_ME_TOPIC_FEAR_AND_GREED:
+                result = await self._get_fear_and_greed_data(session)
+
+            if result:
+                await self._async_notify_consumers(
+                    {
+                        services_constants.FEED_METADATA: topic,
+                    }
+                )
+            await asyncio.sleep(self.API_RATE_LIMIT_SECONDS)
+        await asyncio.sleep(self._get_sleep_time_before_next_wakeup())
+
+    async def _update_loop(self):
+        async with aiohttp.ClientSession() as session:
+            while not self.should_stop:
+                try:
+                    await self._push_update_and_wait(session)
+                except Exception as e:
+                    self.logger.exception(e, True, f"Error when receiving Alternative.me data: ({e})")
+                    self.should_stop = True
+            return False
+
+    async def _start_service_feed(self):
+        try:
+            asyncio.create_task(self._update_loop())
```

Comment:
I think we should save the handle to this task and cancel it when the feed is stopped

Reply:
Right! Do you want me to also update the existing google feed that has the same code?

Reply:
😮 yes please